service_capacity_modeling/models/org/netflix/kafka.py

```diff
@@ -76,14 +78,27 @@ def _estimate_kafka_requirement(
     needed_cores = sqrt_staffed_cores(normalized_to_mib)
 
     # (Nick): Keep 40% of available bandwidth for node recovery
-    needed_network_mbps = simple_network_mbps(desires) * 1.40
+    # (Joey): For kafka BW = BW_write + BW_reads
+    #   let X = input write
+    #   BW_write = [BW_copies] + BW_leader
+    #   BW_write = [2X * (RF - 1)] + X
+    #   BW_write = 2X * RF - 2X + X = 2X * RF - X
+    #   BW_write = X * (2RF - 1)
```

Comment:
I am trying to understand why scale up by `2RF - 1` here. This would mean `BW_write = X*3` for replication factor of 2

I was expecting `BW_write = X * RF` here as the total write BW expected for the cluster

Reply:
Yeah this math assumes that the first write goes to the right replica. In that case you have X `in` to the first replica, and X `out` + X `in` per replica.

If kafka doesn't route the write to the correct first replica (in C* land we call that "token aware routing"), then the BW is X `in` to the coordinator + `RF * 2 * X` for each replica.

Reply:
I see, thanks for sharing the C* context. For Kafka, the kafka client would always directly write to the right leader replica for all writes, so `Bwrite = X * RF`

Reply:
Fixed per your suggestions